Packs/Citrix/Integrations/CitrixDaasEventCollector/CitrixDaasEventCollector.py

```diff
@@ -0,0 +1,370 @@
+                break
+
+        operations = operations[:limit]
+
+        for operation in operations:
+            operation["_time"] = operation.get("FormattedStartTime")
+        return operations, raw_res
+
+
+""" HELPER FUNCTIONS """
+
+
+def get_events_command(client: Client, args: dict):  # type: ignore
+    limit = int(args.get("limit", "10"))
+    search_date_option = args.get("search_date_option")
+    days = args.get("days")
+
+    should_push_events = argToBoolean(args.get("should_push_events", False))
+
+    demisto.debug(f"Running citrix-daas-get-events with {should_push_events=}")
+
+    operations, raw_res = client.get_operations_with_pagination(limit=limit, search_date_option=search_date_option, days=days)
+
+    results = CommandResults(
+        outputs_prefix="CitrixDaas.Event",
+        outputs_key_field="Id",
+        outputs=operations,
+        readable_output=tableToMarkdown("Events List", operations),
+        raw_response=raw_res,
+    )
+
+    if should_push_events:
+        demisto.debug(f"send {len(operations)} events to xsiam")
+        send_events_to_xsiam(operations, vendor=VENDOR, product=PRODUCT)
+
+    return results
+
+
+def days_since(timestamp_str) -> int:
+    """Returns 0 if the timestamp is less than one hour old; otherwise,
+    returns the day difference (1 = today, 2 = yesterday, etc.)."""
+    # Parse the ISO-8601 timestamp with Zulu time (UTC)
+    dt = datetime.strptime(timestamp_str, API_RES_DATE_FORMAT)
+    dt = dt.replace(tzinfo=timezone.utc)
+
+    # Current time in UTC
+    now = datetime.now(timezone.utc)
+
+    delta = now - dt
+    # Difference in minutes
+    minutes = delta.total_seconds() / 60
+    # Difference in days
+    days = delta.days
+
+    if minutes < 60:
+        return 0
+
+    if days <= 0:
+        return 1
+
+    return delta.days + 1
+
+
+def deduplicate_events(events: list[dict[str, Any]], last_fetched_ids: list[str]) -> list[dict[str, Any]]:
+    """Remove already-processed events based on previously fetched IDs."""
+
+    if not last_fetched_ids:
+        demisto.debug("[Dedup] No deduplication needed (first run - no previous IDs)")
+        return events
+
+    demisto.debug(f"[Dedup] Checking {len(events)} events against {len(last_fetched_ids)} previously fetched IDs")
+
+    # Convert to set for O(1) lookup
+    fetched_ids_set = set(last_fetched_ids)
+
+    # Filter out events that were already fetched
+    new_events = [event for event in events if event.get("id") not in fetched_ids_set]
+
+    skipped_count = len(events) - len(new_events)
+    if skipped_count > 0:
+        demisto.debug(f"[Dedup] Skipped {skipped_count} duplicates. {len(new_events)} new events remain.")
+    else:
+        demisto.debug("[Dedup] No duplicates found.")
+
+    return new_events
+
+
+def fetch_events_command(client: Client, max_fetch: int, last_run: dict):
+    last_run_date = last_run.get("LastRun")
+    last_fetched_ids = last_run.get("LastFechedIds", [])
```

Comment:
```suggestion
    last_fetched_ids = last_run.get("LastFetchedIds", [])
```
How did it work with the typo?